lib/logmsg/tests/test_tags.c

```diff
@@ -82,50 +71,44 @@ test_tags(void)
       name = get_tag_by_id(i);
 
       tag_name = log_tags_get_by_id(i);
-      test_msg("Looking up tag by id %d %s(%s)\n", i, tag_name, name);
-      if (tag_name)
-        {
-          if (!g_str_equal(tag_name, name))
-            test_fail("Bad tag name for id %d %s (%s)\n", i, tag_name, name);
-        }
-      else
-        test_fail("Error looking up tag by id %d %s\n", i, name);
+      cr_log_info("Looking up tag by id %d %s(%s)\n", i, tag_name, name);
+
+      cr_assert_not_null(tag_name, "Error looking up tag by id %d %s\n", i, name);
+      cr_assert_str_eq(name, tag_name, "Bad tag name for id %d %s (%s)\n", i, tag_name, name);
 
       g_free(name);
     }
 
   for (i = NUM_TAGS; i < (NUM_TAGS + 3); i++)
     {
+      cr_log_info("Looking up tag by invalid id %d\n", i);
       tag_name = log_tags_get_by_id(i);
-      test_msg("Looking up tag by invalid id %d\n", i);
-      if (tag_name)
-        test_fail("Found tag name for invalid id %d %s\n", i, tag_name);
+      cr_assert_not(log_tags_get_by_id(i), "Found tag name for invalid id %d %s\n", i, tag_name);
```

Comment:
Optional: calling the 2nd time `log_tags_get_by_id` is not required, as it is already *cached* in `tag_name` variable.